File: Java/Medium/Accepted/merge-intervals_6.java
class Solution {
    public int[][] merge(int[][] intervals) {
        
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        List<int[]> output = new ArrayList<>();
        for (int[] interval : intervals) {
            if (output.isEmpty() || output.get(output.size() - 1)[1] < interval[0]) {
                output.add(interval);
            } else {
                output.get(output.size() - 1)[1] = Math.max(output.get(output.size() - 1)[1], interval[1]);
            }
        }
        return output.toArray(new int[output.size()][]);
    }
}

File: Rust/Medium/CompileError/merge-intervals_6.rs
impl Solution {
    pub fn merge(intervals: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        

        let mut intervals = intervals;
        let mut merged = Vec::new();
        intervals.sort_by_key(|x| x[0]);
        for interval in intervals {
            if merged.is_empty() || merged.last().unwrap()[1] < interval[0] {
                merged.push(interval);
            } else {
                let last = merged.pop().unwrap();
                merged.push(vec![last[0], interval[1].max(last[1])]);
            }
        }
        merged
    }
}